service/builder/exporters_builder.go

```diff
@@ -172,7 +174,7 @@ func (eb *ExportersBuilder) buildExporter(
 		// if there are no pipelines associated with the exporter.
 		eb.logger.Warn("Exporter " + config.Name() +
 			" is not associated with any pipeline and will not export data.")
-		return exporter, nil
+		return nil, nil
```

Comment:
Not sure this is what we want in this case. I think even if an exporter is not used in any pipelines, we still want to build and return it as long as it's specified. WDYT? @pjanotti @tigrannajaryan

Reply:
It seems strange to return nil and nil error. I think that the natural behavior here is to return error instead of log a warning. What is the current behavior if the exporter is explicitly disabled on the config? If `disabled` is not set to true and the exporter is not connected to any pipeline it seems reasonable to error in that case, if disabled is set to `true`, independent of the internal implementation, it seems reasonable to allow the collector to go ahead with perhaps logging a warning.

Reply:
@pjanotti , the current behavior is that if an exporter is enabled (which is the default) and not associated with any pipeline, exiting the agent will causes a nil pointer panic.

- I added a condition to check if the exporter is enabled in the updated PR.

Reply:
I agree with @songy23 and @pjanotti 
We already give warning on exporter that is not used in a pipeline. I don't see a need to make this an error.
If this causes a panic later we need to fix that panic, not make it an erroneous config.

Reply:
Hi, @tigrannajaryan , thanks for your comment. The panic is caused by having a non-nil exporter with nil `stopFunc` [1], if the exporter is not used by any pipeline.

If I understand @pjanotti 's suggestion correctly, `` If disabled is not set to true and the exporter is not connected to any pipeline it seems reasonable to error in that case``. This is exactly what the PR does.

[1] https://github.com/open-telemetry/opentelemetry-service/blob/f20385bac8e7a3db68a1f8cfeaa3a3edcde9f419/service/builder/exporters_builder.go#L178-L206